versioned_docs/version-1.x/docs/recipes/protected-app/README.mdx

```diff
@@ -0,0 +1,137 @@
+
+Another way to secure your origin server is by using JSON Web Tokens (JWT).
+
+Each authed request from the Protected App includes the following header:
+
+```
+Logto-ID-Token: <JWT>
+```
+
+The JWT is called [ID Token](https://openid.net/specs/openid-connect-core-1_0.html#CodeIDToken) which is signed by Logto and contains user information. By validating this JWT, you can confirm the request comes from the Protected App and deny any requests that do not include this header.
+
+The token is encrypted and signed as a [JWS](https://datatracker.ietf.org/doc/html/rfc7515) token.
+
+The validation steps:
+
+1. [Validating a JWT](https://datatracker.ietf.org/doc/html/rfc7519#section-7.2)
+2. [Validating the JWS signature](https://datatracker.ietf.org/doc/html/rfc7515#section-5.2)
+3. The token's issuer is `https://<your-logto-domain>/oidc` (issued by your Logto auth server)
+
+```js
+const express = require('express');
+const jwksClient = require('jwks-rsa');
+const jwt = require('jsonwebtoken');
+
+const ISSUER = 'https://<your-logto-domain>/oidc';
+const CERTS_URL = 'https://<your-logto-domain>/oidc/jwks';
+
+const client = jwksClient({
+  jwksUri: CERTS_URL,
+});
+
+const getKey = (header, callback) => {
+  client.getSigningKey(header.kid, function (err, key) {
+    callback(err, key?.getPublicKey());
+  });
+};
+
+const verifyToken = (req, res, next) => {
+  const token = req.headers['Logto-ID-Token'];
+
+  // Make sure that the incoming request has our token header
+  if (!token) {
+    return res
+      .status(403)
+      .send({ status: false, message: 'missing required Logto-ID-Token header' });
+  }
+
+  jwt.verify(token, getKey, { iss: ISSUER }, (err, decoded) => {
+    if (err) {
+      return res.status(403).send({ status: false, message: 'invalid id token' });
+    }
+
+    req.user = decoded;
+    next();
+  });
+};
+
+const app = express();
+
+app.use(verifyToken);
+
+app.get('/', (req, res) => {
+  res.send('Hello World!');
+});
+
+app.listen(3000);
+```
+
+## Get authentication state and user information
+
+If you need to get authentication and user information for your application, you can also use the `Logto-ID-Token` header.
+
+## Customize authentication rules
+
+By default, the Protected App will protect all routes. If you need to customize the authentication rules, you can set the "Custom authentication rules" field in Console.
+
+It supports regular expressions, here are two case scenarios:
+
+1. To only protect routes `/admin` and `/privacy` with authentication: `^/(admin|privacy)/.*`
+2. To exclude JPG images from authentication: `^(?!.*\.jpg$).*$`
+
+## Transition to SDK integration
+
+The Protected App is designed to simplify the authentication process. However, if you decide to transition to SDK integration for better control and customization, you can reuse the application configs. The Protected App is actually a "Traditional Web App" in Logto, you can find the "AppId" and "AppSecret" in the application settings.
```

Comment:
Should we encourage people to create a new app instead of reuse the protected app?

Reply:
Updated, "reuse" is for smooth transition only